code/modules/reagents/reagent_containers.dm

```diff
@@ -180,6 +180,35 @@
 				reagent_overlay.overlays += overlay_image(icon, reagent.reagent_overlay, reagent.color, RESET_COLOR | RESET_ALPHA)
 	return reagent_overlay
 
+/obj/item/chems/ProcessAtomTemperature()
+
+	. = ..()
+
+	if(QDELETED(src) || !reagents?.total_volume || !ATOM_IS_OPEN_CONTAINER(src) || !isatom(loc))
+		return
+
+	// Vaporize anything over its boiling point.
+	for(var/reagent in reagents.reagent_volumes)
+		var/decl/material/mat = GET_DECL(reagent)
+		if(!isnull(mat.boiling_point) && temperature >= mat.boiling_point)
+			// TODO: reduce atom temperature?
+			var/removing = min(5, reagents.reagent_volumes[reagent])
+			reagents.remove_reagent(reagent, removing, defer_update = TRUE, removed_phases = MAT_PHASE_LIQUID)
+			loc.take_vaporized_reagent(reagent, removing)
+			. = null // Don't return PROCESS_KILL, we may have more to vaporise
+	reagents.handle_update()
+
+/obj/item/chems/take_vaporized_reagent(reagent, amount)
+	// TODO: check ATOM_IS_OPEN_CONTAINER(src) when there is a closed container for alembic vessels
```

Comment:
as you mentioned on discord, this comment may not apply anymore? should look into that